test/jdk/java/lang/reflect/Proxy/DefaultMethods.java

```diff
@@ -371,16 +379,72 @@ public void testIllegalArgument(Object... args) throws Exception {
     public void testInvocationTargetException(Throwable exception) throws Exception {
         ClassLoader loader = DefaultMethods.class.getClassLoader();
         IX proxy = (IX)Proxy.newProxyInstance(loader, new Class<?>[]{IX.class}, HANDLER);
+        InvocationHandler2.SuperInvoker superInvoker = getSuperInvoker(proxy);
         Method m = IX.class.getMethod("doThrow", Throwable.class);
         try {
-            InvocationHandler.invokeDefaultMethod(proxy, m, exception);
+            superInvoker.invokeSuper(proxy, m, exception);
         } catch (InvocationTargetException e) {
             assertEquals(e.getCause(), exception);
         }
     }
 
-    private static final InvocationHandler HANDLER = (proxy, method, params) -> {
+    private static InvocationHandler2.SuperInvoker getSuperInvoker(Object proxy) {
+        try {
+            Class<?> proxyClass = proxy.getClass();
+            Class<?>[] intfcs = proxyClass.getInterfaces();
+            // take any method of any interface with a method
+            Method m = Stream.of(intfcs)
+                  .flatMap(intf -> Stream.of(intf.getMethods()))
+                  .findAny().orElseThrow(() -> new IllegalArgumentException("proxy has no methods"));
+            Object returnValue = getDefault(m.getReturnType());
+            Class<?>[] ptypes = m.getParameterTypes();
+            Object[] args = new Object[ptypes.length];
+            for (int i = 0; i < ptypes.length; i++) {
+                args[i] = getDefault(ptypes[i]);
+            }
+            InvokerCapturingHandler handler = new InvokerCapturingHandler(returnValue);
+            // we assume caching works and we get the same proxy class
+            Object captureProxy = Proxy.newProxyInstance(proxyClass.getClassLoader(), intfcs, handler);
+            m.invoke(captureProxy, args);
+            return handler.getSuperInvoker();
+        } catch (Exception e) {
+            throw new AssertionError("Can't get Proxy lookup", e);
+        }
```

Comment:
Ops, forgot to change the exception message. Should be: "Can't get SuperInvoker"